.github/workflows/build.yml

```diff
@@ -41,12 +41,12 @@ jobs:
       - name: Setup PNPM
         uses: pnpm/action-setup@v2
         with:
-          version: 7.1.5
+          version: 8
 
       - name: Setup NodeJS
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v4
         with:
-          node-version: 16.x
+          node-version: 18
```

Comment:
this should be the node version recommended for studio-client 2110.2 - whatever that is (but I suppose it is 16) - or is node always 100% backwards compatible?

Reply:
node 18 everywhere in the CM universe meanwhile.
I think we should aim to get this plugin working together with 2310 and master.
but yeah, it also should work together with older versions if possible.

@pitgrap maybe any thoughts?

Reply:
node16 is out of support. 

My recommendation:
- one major plugin release per major cmcc release. 
- for every cmcc major release always just support the latest LTS version:
  - CMCC 10 = 2107.X (node16 + java7?)
  - CMCC 11 = 2310.x (node18 + java11)
  - CMCC 12 = master (probably node20 + java17)

I don't know, how we could otherwise guarantee if it works. 
Most npm third-party libs strip support for old node releases.

If we don't care, we should at least test java and node versions via matrix in github action...

Reply:
@mkleine What do you think?

Reply:
I second Michael's recommendation. We should keep the main branch of the plugins up to date with respect of build and runtime dependencies. That's basically what we decided to do with the commerce adapters as well: Have a major version that is compatible with at least one CMCC major version. We try not to break the API, so we can tell customers that they can use any compatible version of their liking, be it the first one (built with old tooling, referencing the eldest compatible CM API) or a newer one (built with new tooling, referencing the current CM API).

Reply:
after discussing this with (@fwienber was also involved) we will go back to the original cms version.

We want to stay backward compatible as long as possible.
All cms artifacts are provided by the runtime environment so we only need to bump the 3rd party libs which are packaged with the plugin. Since we do not use new/updated cms api in this pr.
As long as the plugin runs on newer versions we can avoid maintenance branches this way.